core/connector/src/main/java/org/exoplatform/wcm/connector/collaboration/OpenInOfficeConnector.java

```diff
@@ -183,28 +189,40 @@ public Response updateDocumentTitle(
    * @param request
    * @param filePath
    * @return
-   * @throws Exception
    */
   @GET
+  @RolesAllowed("users")
   @Path("/checkout")
   public Response checkout(@Context Request request,
                            @QueryParam("filePath") String filePath,
                            @QueryParam("workspace") String workspace
-  ) throws Exception {
-    Session session = WCMCoreUtils.getSystemSessionProvider().
-        getSession(workspace, WCMCoreUtils.getRepository());
-    Node node = (Node)session.getItem(filePath);
-
-    if(node.canAddMixin(VERSION_MIXIN)){
-      node.addMixin(VERSION_MIXIN);
-      node.save();
-      node.checkin();
-      node.checkout();
-    }
+  ) {
+    try {
+      Node node = null;
+      try {
+        node = (Node) nodeFinder.getItem(workspace, filePath);
+      } catch (PathNotFoundException e) {
+        node = (Node) nodeFinder.getItem(workspace, Text.unescapeIllegalJcrChars(filePath));
+      }
 
-    if(!node.isCheckedOut()) node.checkout();
+      if (node.canAddMixin(VERSION_MIXIN)) {
+        node.addMixin(VERSION_MIXIN);
+        node.save();
+        node.checkin();
+        node.checkout();
+      }
 
-    return Response.ok(String.valueOf(node.isCheckedOut()), MediaType.TEXT_PLAIN).build();
+      if (!node.isCheckedOut()) {
+        node.checkout();
+      }
+
+      return Response.ok(String.valueOf(node.isCheckedOut()), MediaType.TEXT_PLAIN).build();
+    } catch (PathNotFoundException | AccessDeniedException e) {
+      return Response.status(Status.NOT_FOUND).build();
```

Comment:
Shouldn't we log something in this case ?

Reply:
done